### ISSUE TITLE
Fix S1186 FP: Empty method comes from interface

### ISSUE THREAD

**Issue by hankovich** [SonarSource/sonar-dotnet#7629]
### Description

S1186 is reported for empty methods that implement interface

### Repro steps

```c#
interface IFilter
{
    void OnEnter();

    void OnExit();
}

class Filter : IFilter
{
    public void OnEnter() // <- is reported here
    {
    }

    public void OnExit()
    {
        Console.WriteLine("exiting");
    }
}
```

### Expected behavior

No diagnostic is reported

### Actual behavior

Diagnostic is reported

### Related information

* C#/VB.NET Plugins version 9.5.0.73987
* Visual Studio version latest preview
* MSBuild / dotnet version .NET 8 preview 6

**Comment by cristian-ambrosini-sonarsource** [comment]
Hi @hankovich, thank you for reporting this. I confirmed it as False Positive.
